### DISCUSSION TITLE
Javascript KNX function video showoff

### DISCUSSION THREAD

**Discussion by Supergiovane** [Supergiovane/node-red-contrib-knx-ultimate#373] [Show and tell]
Node-Red KNX Ultimate 3.1.x introduces a powerful feature, which allows you to write Javascript code to handle input and output messages. In this example, a device that would emit a numeric value (lux) to the flow, becomes a boolean (true/false) that turns the connected light on or off.

https://github.com/user-attachments/assets/e428cb04-6aa5-43a6-b3c8-b30e25ec9eff

**Comment by uschmelmer** [comment]
Hi,
what's the advantage over a function node placed afterwards?
This adds complexity to the knx-ultimate package is less modular in my opinion.

Anyways, keep up the great work with knx-ultimate! Thank you very much for it!
Kind Regards,
uschmelmer

**Comment by Supergiovane** [comment]
Hi
You can evaluate other GA in real time (getGAValue function) prior to pass the message. Imagine you have a "toggle" light switch, where you must evaluate the status GA prior to send the on/off command.
You can control other GA in real time (setGAValue function) as well.
You can also use basic coding like timers, to switch on the light and then switch off it after some time, just in the KNX node.
You can simplify your flow.
Obviously, you need to have knowledge of javascript. It's very powerful.

**Comment by uschmelmer** [comment]
Wow just tested it.
really cool, especially the support for Timers is a really great value!

Thanks!
